[WIP] doc: add MAINTAINERS.md (#26868)
bitcoin/bitcoin: MAINTAINERS.md

# Maintainers

This document contains list of present maintainers, their role and process to add new maintainers.

> ## Current Maintainers

Reviewer: I'm unsure if this section is a good idea, though perhaps not for a good reason.  In general, people (e.g. journalists and non-contributors to the project) in my experience overestimate the role of maintainers (i.e. as power/control/gatekeepers rather than implementing the rough consensus of reviewers), and conversely, they undervalue the key vital role of the contributors who are doing consistent review and testing that shape and determine rough consensus.

Also, these roles may be fluid, require updating here, and "General" before each seems redundant.

Author: - Removed 'Focus Area' column 
- I understand your point about journalists however they would anyway find these names from other links which maybe misleading, outdated etc. Role of maintainers and contributors is defined in next section with 'rough consensus' part mentioned. A `maintainers.md` file without names of maintainers would be incomplete IMO

|   Username    |    Focus Area   |
| ------------- | -------------   |
|  MarcoFalke   | General, QA     |
|  fanquake     | General, Build  |
|  hebasto      | General, UI/UX  |
|  achow101     | General, Wallet |
|  glozow       | General, Mempool|

## Responsibilities and roles

Project maintainers have commit access and are responsible for merging patches
from contributors. They perform a janitorial role merging patches that the team

> agrees should be merged. Maintainers make sure that patches are safe and align

Reviewer: perhaps: "They perform a janitorial role of implementing, by merging patches, the rough consensus of reviewers. Maintainers also make sure..."

Author: I used this sentence, let me know if it conveys the correct meaning: 

"They perform a janitorial role of implementing the rough consensus of reviewers by merging patches."

with the project goals. However, their decision is not the last step, as patches
must also go through post-merge review and testing before being implemented.

> The maintainers’ role is by agreement of project contributors.

Reviewer: ```suggestion
The maintainers’ role is by rough consensus of project contributors.
```

Author: Done

## Self Merges

In general, it is considered best practice for maintainers to avoid merging their
own changes directly into the codebase. This is because merging your own code can
create a potential conflict of interest, where a maintainer's personal interests
may override the best interests of the project.

> However, it is acceptable for maintainers to merge their own changes if pull request

Reviewer: ```suggestion
However, it may be acceptable for maintainers to occasionally merge their own changes if the pull request
```

has enough reviews and no other maintainer is available to merge.

> ## Nomination and Adding New Maintainers

Reviewer: ```suggestion
## Nomination and Addition of New Maintainers
```

Author: Done

> The current maintainers or contributors can nominate a long term contributor

Reviewer: I think the word "contributor" includes maintainers.
```suggestion
Contributors can nominate a long-term contributor
```

Author: Done

in the [IRC channel](https://web.libera.chat/#bitcoin-core-dev) to take on the

> role of a maintainer. Contributors are free to self nominate as well.

Reviewer: ```suggestion
role of a maintainer. Contributors are free to self-nominate as well.
```

Author: Done

> Maintainers are added by a new maintainer opening a pull-request to add their

Reviewer: ```suggestion
Maintainers are added by a new maintainer opening a pull request to add their
```

Author: Done

key to the Trusted Keys file. For example, see [this pull
request](https://github.com/bitcoin/bitcoin/pull/23798). The adding of the key

> is reviewed by the community and merged by one of the maintainer. Before opening

Reviewer 2: Typo: "maintainer" should be "maintainers"

Author: Fixed

> is reviewed by the community and merged by one of the maintainers. Before opening

Reviewer: ```suggestion
is reviewed by the community and merged by one of the maintainers based on rough consensus of the reviewing contributors. Before opening
```

Author: Done

such a pull request, it is typical that the adding
of the maintainer might be discussed in the regular Bitcoin Core IRC meeting,

> but the actual decision making process occurs in the pull request.

Reviewer: ```suggestion
but the actual decision-making process occurs in the pull request.
```

Author: Done

## Maintainers "Moving On"

Maintainers may choose to cease being a maintainer at any time without notice.
Where possible, maintainers should try to gracefully transition out of their
role, until a new maintainer has been selected to replace them (if needed).

## Maintainer Removal

There is currently no established procedure for removing a maintainer. If it is
deemed necessary to remove a maintainer, it should be discussed openly among the

> other maintainers and contributors and a decision should be made through a

Author: Done

# Maintainers

This document contains list of present maintainers, their role and selection process.

## Current Maintainers

|   Username    |    Focus Area   |
| ------------- | -------------   |
|  MarcoFalke   | General, QA     | 
|  fanquake     | General, Build  |
|  hebasto      | General, UI/UX  |
|  achow101     | General, Wallet |
|  glozow       | General, Mempool|

## Responsibilities and roles

Project maintainers have commit access and are responsible for merging patches 
from contributors. They perform a janitorial role merging patches that the team 

> agrees should be merged. They also act as a final check to ensure that patches 

Reviewer 3: A maintainer decision isn't final. It is merely a piece before post-merge review and post-merge testing happens. Also, there is a release process and the every user is free to use their own judgement whether and what software they want to run.

Author: I copied this part from https://bitcoincore.org/en/about/. Rephrased 'final check' and added a sentence based on your suggestion.

are safe and in line with the project goals. The maintainers’ role is by agreement 
of project contributors.

## Self Merges

In general, it is considered best practice for maintainers to avoid merging their 
own changes directly into the codebase. This is because merging your own code can 
create a potential conflict of interest, where a maintainer's personal interests 
may override the best interests of the project.

However, it is acceptable for maintainers to merge their own changes if pull request
has enough reviews and no other maintainer is available to merge.

## Nomination and Selection of Maintainers

> The current maintainers or contributors can open a 'call for maintainers' issue. 

Reviewer 3: I don't think a single maintainer in the past has been picked based on such a thread, so it seems weird to start prescribing it now and mentioning it seems misleading.

Author: Removed 'call for maintainers' section and replaced by nomination in IRC channel. However, it was suggested by @ariard as an improvement that I agree with and could be discussed in next IRC meeting. It can be added later if enough contributors agree with it.
